### PR TITLE
SP2-86 create asset manager class

### PR DESCRIPTION
**Pull request by magwrap**
# Asking for review | NOT TESTED

I think basic functionallity is done. Tests are still not ready! But if you need this class for further work feel free to emergency merge into dev or your local branch(may result in conflicts).

 - provides data about production units and heating grid
 - multiple overloaded ways for writing (editing) production unit 
 - JsonController does basic Seriallization, Deserializaton, readdata and writedata implementing IReadWriteController and IDeserializeSerialize
 - also implemented JsonDataSource (followed the convension followed in #12) but for me using JsonController makes more sense so JsonDataSource is unused.

 
 _Question_:
 Isin't the name HeatingUnit kinda confusing? It implies existance of other sub-group of units in ProductionUnits but in reality they are the same.

### CODE REVIEW

**Review by pkyeager — CHANGES_REQUESTED** [review]
Good job, if tested(simple console.log, checking if units get instantiated, access is working and etc.) or directly a unit test. 

Are more md files for documentation in coming?

**Comment by pkyeager** [comment]
> _Question_: Isin't the name HeatingUnit kinda confusing? It implies existance of other sub-group of units in ProductionUnits but in reality they are the same.

I think, this is a legit question.

**Comment by magwrap** [comment]
> Good job, if tested(simple console.log, checking if units get instantiated, access is working and etc.) or directly a unit test.
> 
> Are more md files for documentation in coming?

Yes of course documentation is coming too, I did the logs in my project but deleted them so not to pollute the code.

File: Heatington/Controllers/JsonController.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        catch (Exception e)
+        {
+            Utilities.DisplayException($"Unknown exception occured: {e.Message}");
+            throw;
+        }
+    }
+}
+
+// WORK IN PROGRESS
+public class ProductionUintJsonConverter : JsonConverter<ProductionUnit>
```
Comment (KonyakB) [inline Heatington/Controllers/JsonController.cs:84]:
```suggestion
public class ProductionUnitJsonConverter : JsonConverter<ProductionUnit>
```
there is a typo here ("ProductionUint...")

**Review by KonyakB — CHANGES_REQUESTED** [review]
seems good beside the typo

**Review by pkyeager — APPROVED** [review]

**Review by frske — APPROVED** [review]
LGTM

**Review by KonyakB — APPROVED** [review]

**Pull request merged**